### PR TITLE
feat: cart-management

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
- add to cart
- remove from cart
- increase count
- calculate total price

Screenshot (68)

### CODE REVIEW

File: src/pages/CartPage/Cart.jsx [hunk 1]
```diff
@@ -1,6 +1,142 @@
...
+        cart.map((item) =>
+          item._id === id ? { ...item, quantity: item.quantity + 1 } : item
+        )}
+      );
+    };
+  
+    const decreaseCountHandler = (id) => {
+        dispatch({ type:"PRODUCT_COUNT", payload:
+        cart.map((item) =>
+          item._id === id ? { ...item, quantity: item.quantity - 1 } : item
```
Comment (vinc3nati) [inline src/pages/CartPage/Cart.jsx]: very neat and organized code, just one suggestion handle the `decreaseCountHandler` case when count is less than 1.

**Review by vinc3nati — APPROVED** [review]

File: src/pages/CartPage/Cart.jsx [hunk 2]
```diff
@@ -1,6 +1,142 @@
...
+        )}
+      );
+    };
```
Comment (romabulani) [inline src/pages/CartPage/Cart.jsx]: For increment and decrement you can call the api of mockbee

File: src/pages/CartPage/Cart.jsx [hunk 3]
```diff
@@ -1,6 +1,142 @@
...
+      const response = await axios.delete(
+        `/api/user/cart/${productId}`,
+        {
+            headers:{
+              authorization: encodedToken,
+          },
+        }
+      );
+     
+    dispatch({type:"ADD_TO_CART", payload:response.data.cart});
```
Comment (romabulani) [inline src/pages/CartPage/Cart.jsx]: You should always write try and catch for API calls

**Review by romabulani — APPROVED** [review]
- Code is clean
- classnames are self explanatory
- Have given some suggestions, otherwise good to merge

**Pull request merged**